EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc

```diff
@@ -66,7 +70,33 @@ HcalRawToDigi::HcalRawToDigi(edm::ParameterSet const& conf):
   produces<QIE10DigiCollection>();
   produces<QIE11DigiCollection>();
   produces<QIE10DigiCollection>("ZDC");
-  
+
+  // If additional qie10 samples were requested,
+  // decalre that we will produce this collection
+  // with the given tag.  Also store the map
+  // from nSamples to tag for later use
+  for( unsigned idx = 0; idx < saveQIE10DataNSamples_.size(); ++idx ) {
```

Comment:
add a check that `saveQIE10DataNSamples_` and `saveQIE10DataTags_` are the same length, otherwise throw an exception